tests/suite/nginxgateway_test.go

```diff
@@ -0,0 +1,207 @@
+package main
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	core "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
+	"github.com/nginxinc/nginx-gateway-fabric/tests/framework"
+)
+
+var _ = Describe("NginxGateway", Ordered, Label("functional", "nginxGateway"), func() {
+	var (
+		ngfPodName string
+
+		namespace          = "nginx-gateway"
+		nginxGatewayNsname = types.NamespacedName{Name: releaseName + "-config", Namespace: namespace}
+
+		files = []string{
+			"nginxgateway/nginx-gateway.yaml",
+		}
+	)
+
+	AfterAll(func() {
+		// cleanup and restore NGF instance
+		teardown(releaseName)
+		setup(getDefaultSetupCfg())
+	})
+
+	getNginxGateway := func(nsname types.NamespacedName) (ngfAPI.NginxGateway, error) {
+		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetStatusTimeout)
+		defer cancel()
+
+		var nginxGateway ngfAPI.NginxGateway
+
+		if err := k8sClient.Get(ctx, nsname, &nginxGateway); err != nil {
+			return nginxGateway, err
+		}
+
+		return nginxGateway, nil
+	}
+
+	verifyAndReturnNginxGateway := func(nsname types.NamespacedName) ngfAPI.NginxGateway {
+		nginxGateway, err := getNginxGateway(nsname)
+		Expect(nginxGateway).ToNot(BeNil())
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(nginxGateway.Status.Conditions).To(HaveLen(1))
+		condition := nginxGateway.Status.Conditions[0]
+
+		Expect(condition.Type).To(Equal("Valid"))
+		Expect(condition.Reason).To(Equal("Valid"))
+
+		return nginxGateway
+	}
+
+	When("testing NGF on startup", func() {
+		BeforeEach(func() {
+			teardown(releaseName)
+
+			ns := &core.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: namespace,
+				},
+			}
+			Expect(resourceManager.Apply([]client.Object{ns})).To(Succeed())
+		})
```

Comment:
We don't need to do this on the first test run, because NGF will already exist. Tearing down just to start back up again adds unnecessary time to the test run.

Reply:
Well since we've already defined `Ordered` in this test suite, we've made it clear that order matters, so adding a new test needs to abide by this.

We can also leave a comment in the first test that says we aren't tearing down because it already exists from the suite setup.

Reply:
Ok, I agree with the first point, though the second point, won't this test get affected by other functional tests? -- meaning the NGF instance could have existed through the running of all the other functional tests, not straight from the suite setup?

Reply:
Yes, but unless those tests are changing the log level (which they shouldn't be) then they shouldn't impact this test. We're still looking for the `debug` log message that is only configured when the NginxGateway resource has it set. So regardless of how NGF was running beforehand, that functionality is still tested properly here.

Reply:
Now, if we get into a situation in the future where our tests are more complex and NginxGateway has more fields in it that may be altered (or the resource is deleted), then yes, we will likely need to make this more independent. But I don't think we need to worry about that right now.